File: ExpressionTreeTestObjects/CSCompiler/Lambda.cs
using System;
using System.Linq.Expressions;
using static ExpressionTreeTestObjects.Categories;
using static ExpressionTreeTestObjects.Functions;

namespace ExpressionTreeTestObjects {
    partial class CSCompiler {
        [Category(Lambdas)]
        internal static readonly Expression NoParametersVoidReturn = Expr(() => Console.WriteLine());
        
        [Category(Lambdas)]
        internal static readonly Expression OneParameterVoidReturn = Expr((string s) => Console.WriteLine(s));
        
        [Category(Lambdas)]
        internal static readonly Expression TwoParametersVoidReturn = Expr((string s1, string s2) => Console.WriteLine(s1 + s2));

        [Category(Lambdas)]
        internal static readonly Expression NoParametersNonVoidReturn = Expr(() => "abcd");

        [Category(Lambdas)]
        internal static readonly Expression OneParameterNonVoidReturn = Expr((string s) => s);

        [Category(Lambdas)]
        internal static readonly Expression TwoParametersNonVoidReturn = Expr((string s1, string s2) => s1 + s2);
    }
}

File: ExpressionTreeTestObjects.VB/VBCompiler/Lambda.vb
Partial Module VBCompiler
    <Category(Lambdas)>
    Friend NoParametersVoidReturn As Expression = Expr(Sub() Console.WriteLine())

    <Category(Lambdas)>
    Friend OneParameterVoidReturn As Expression = Expr(Sub(s As String) Console.WriteLine(s))

    <Category(Lambdas)>
    Friend TwoParametersVoidReturn As Expression = Expr(Sub(s1 As String, s2 As String) Console.WriteLine(s1 + s2))

    <Category(Lambdas)>
    Friend NoParametersNonVoidReturn As Expression = Expr(Function() "abcd")

    <Category(Lambdas)>
    Friend OneParameterNonVoidReturn As Expression = Expr(Function(s As String) s)

    <Category(Lambdas)>
    Friend TwoParametersNonVoidReturn As Expression = Expr(Function(s1 As String, s2 As String) s1 + s2)
End Module